### ISSUE TITLE
My project

### ISSUE THREAD

**Issue by KariaknightX** [KariaknightX/Showing-financial-and-import-csv-file-#1]
import tkinter as tk
from tkinter import messagebox
from tkinter import ttk
from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from matplotlib.animation import FuncAnimation
from PIL import Image, ImageTk
import time

# Reading data
df = pd.read_csv("C:\\Users\\kmokh\\Desktop\\پرو\\heart.csv")

# Main Tkinter application window
app = tk.Tk()
app.title("Financial Management Dashboard")
app.geometry("1000x700")
app.configure(bg='white')

# Adding a background image
background_image = Image.open(r"C:\Users\kmokh\Downloads\techno-background-ln6ntpa0p41e0yy9.jpg")
background_photo = ImageTk.PhotoImage(background_image)
background_label = tk.Label(app, image=background_photo)
background_label.place(x=0, y=0, relwidth=1, relheight=1)

# Style for buttons
button_style = {
    'font': ('Helvetica', 12, 'bold'),
    'bg': '#2C3E50',
    'fg': 'white',
    'activebackground': '#34495E',
    'relief': 'raised',
    'borderwidth': 1,
    'width': 40,
    'height': 2
}

# Frame to hold plots
plot_frame = tk.Frame(app, bg='white')
plot_frame.pack(fill=tk.BOTH, expand=True)

def clear_plot_frame():
    for widget in plot_frame.winfo_children():
        widget.destroy()

def animate_income_achievement(i, ax):
    ax.clear()
    ax.plot(df["Year of Income Achievement"][:i], df["Income"][:i], marker="o", color="green", label="Income")
    ax.set_xlabel("Year")
    ax.set_ylabel("Income")
    ax.set_title("Income Achievement Over Time")
    ax.grid(True)
    ax.legend()

def plot_income_achievement():
    clear_plot_frame()
    fig, ax = plt.subplots(figsize=(8, 5))
    canvas = FigureCanvasTkAgg(fig, master=plot_frame)
    canvas.get_tk_widget().pack(fill=tk.BOTH, expand=True)
    ani = FuncAnimation(fig, animate_income_achievement, fargs=(ax,), frames=len(df), repeat=False)
    canvas.draw()

def animate_spending_evolution(i, ax):
    ax.clear()
    ax.plot(df["Year of Income Achievement"][:i], df["Expense"][:i], marker="o", color="teal", linestyle="--", linewidth=1.5, label="Spending")
    ax.set_xlabel("Year")
    ax.set_ylabel("Spending")
    ax.set_title("Spending Evolution Over Time")
    ax.grid(True, linestyle=":", alpha=0.5)
    ax.legend()

def plot_spending_evolution():
    clear_plot_frame()
    fig, ax = plt.subplots(figsize=(8, 5))
    canvas = FigureCanvasTkAgg(fig, master=plot_frame)
    canvas.get_tk_widget().pack(fill=tk.BOTH, expand=True)
    ani = FuncAnimation(fig, animate_spending_evolution, fargs=(ax,), frames=len(df), repeat=False)
    canvas.draw()

def animate_income_evolution(i, ax):
    ax.clear()
    ax.plot(df["Company Name"][:i], df["Income"][:i], marker="o")
    ax.set_xlabel("Company Name")
    ax.set_ylabel("Income")
    ax.set_title("Income of Each Institute")
    ax.grid(True)

def plot_income_evolution():
    clear_plot_frame()
    fig, ax = plt.subplots(figsize=(8, 5))
    canvas = FigureCanvasTkAgg(fig, master=plot_frame)
    canvas.get_tk_widget().pack(fill=tk.BOTH, expand=True)
    ani = FuncAnimation(fig, animate_income_evolution, fargs=(ax,), frames=len(df), repeat=False)
    canvas.draw()

def combined_analysis_and_budget():
    clear_plot_frame()
    
    # Analysis and Budget Calculation
    total_income = df["Income"].sum()
    total_expense = df["Expense"].sum()
    balance = total_income - total_expense
    average_income = df["Income"].mean()
    average_expense = df["Expense"].mean()

    analysis_text = (
        f"Average Income: {average_income}\n"
        f"Average Expense: {average_expense}\n"
        f"Total Income: {total_income}\n"
        f"Total Expense: {total_expense}\n"
        f"Current Balance: {balance}\n"
    )

    if balance < 100:
        messagebox.showwarning("Warning", "Low balance!")

    # Predict the most successful company type
    company_types = df["Company Type"].unique()
    success_scores = {}

    for company_type in company_types:
        type_df = df[df["Company Type"] == company_type]
        X_type = type_df["Year of Income Achievement"].values.reshape(-1, 1)
        y_type_income = type_df["Income"].values
        y_type_expense = type_df["Expense"].values

        income_model_type = LinearRegression()
        income_model_type.fit(X_type, y_type_income)

        expense_model_type = LinearRegression()
        expense_model_type.fit(X_type, y_type_expense)

        future_years = np.arange(df["Year of Income Achievement"].max() + 1, df["Year of Income Achievement"].max() + 11).reshape(-1, 1)
        predicted_income_type = income_model_type.predict(future_years)
        predicted_expense_type = expense_model_type.predict(future_years)

        total_predicted_income = predicted_income_type.sum()
        total_predicted_expense = predicted_expense_type.sum()
        success_score = total_predicted_income - total_predicted_expense
        success_scores[company_type] = success_score
    
    most_successful_type = max(success_scores, key=success_scores.get)

    analysis_text += f"The most successful company type in the next 10 years will be: {most_successful_type}"

    messagebox.showinfo("Analysis & Budget", analysis_text)

    # Plotting the distribution of company types
    type_counts = df["Company Type"].value_counts()
    fig, ax = plt.subplots(figsize=(8, 5))
    ax.pie(type_counts, labels=type_counts.index, autopct="%1.1f%%", startangle=90)
    ax.set_title("Distribution of Company Types")
    ax.axis('equal')  # Equal aspect ratio ensures that the pie is drawn as a circle.

    canvas = FigureCanvasTkAgg(fig, master=plot_frame)
    canvas.get_tk_widget().pack(fill=tk.BOTH, expand=True)
    canvas.draw()

def predict_income():
    clear_plot_frame()
    
    X = df["Year of Income Achievement"].values.reshape(-1, 1)
    y_income = df["Income"].values

    income_model = LinearRegression()
    income_model.fit(X, y_income)

    future_years = np.arange(df["Year of Income Achievement"].max() + 1, df["Year of Income Achievement"].max() + 11).reshape(-1, 1)
    predicted_income = income_model.predict(future_years)

    fig, ax = plt.subplots(figsize=(8, 5))
    ax.plot(df["Year of Income Achievement"], df["Income"], marker="o", color="green", label="Actual Income")
    ax.plot(future_years, predicted_income, marker="x", color="blue", linestyle="--", label="Predicted Income")
    ax.set_xlabel("Year")
    ax.set_ylabel("Income")
    ax.set_title("Income Prediction for the Next 10 Years")
    ax.legend()
    ax.grid(True)

    canvas = FigureCanvasTkAgg(fig, master=plot_frame)
    canvas.get_tk_widget().pack(fill=tk.BOTH, expand=True)
    canvas.draw()

def predict_expense():
    clear_plot_frame()
    
    X = df["Year of Income Achievement"].values.reshape(-1, 1)
    y_expense = df["Expense"].values

    expense_model = LinearRegression()
    expense_model.fit(X, y_expense)

    future_years = np.arange(df["Year of Income Achievement"].max() + 1, df["Year of Income Achievement"].max() + 11).reshape(-1, 1)
    predicted_expense = expense_model.predict(future_years)

    fig, ax = plt.subplots(figsize=(8, 5))
    ax.plot(df["Year of Income Achievement"], df["Expense"], marker="o", color="red", label="Actual Expense")
    ax.plot(future_years, predicted_expense, marker="x", color="purple", linestyle="--", label="Predicted Expense")
    ax.set_xlabel("Year")
    ax.set_ylabel("Expense")
    ax.set_title("Expense Prediction for the Next 10 Years")
    ax.legend()
    ax.grid(True)

    canvas = FigureCanvasTkAgg(fig, master=plot_frame)
    canvas.get_tk_widget().pack(fill=tk.BOTH, expand=True)
    canvas.draw()

# Adding buttons to the app
tk.Button(app, text="Income Achievement", command=plot_income_achievement, **button_style).pack(pady=3)
tk.Button(app, text="Spending Evolution", command=plot_spending_evolution, **button_style).pack(pady=3)
tk.Button(app, text="Income of Each Institute", command=plot_income_evolution, **button_style).pack(pady=3)
tk.Button(app, text="Analysis, Budget & Predict Successful Company\n & Show Company Type Distribution", command=combined_analysis_and_budget, **button_style).pack(pady=3)
tk.Button(app, text="Predict Income", command=predict_income, **button_style).pack(pady=3)
tk.Button(app, text="Predict Expense", command=predict_expense, **button_style).pack(pady=3)

# Running the Tkinter application
app.mainloop()
